### PR TITLE
Documentation Additions and Changes

### PR DESCRIPTION
**Pull request by halotroop2288**
Here are my contributions to the documentation. I started by adding a page for the *Platform and Plugins* preferences page.

## Done:
- External Services
- Platform and Plugins
  - Experimental Features
  - Update Settings
- SDKs/Bundles/Tools

---

There's more to come!

(P.S. I ***highly recommend*** squashing this PR. It has a *very* messy Git history.)

### CODE REVIEW

File: docs/platform/settings/platform.and.plugins/update.settings.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+![Update Channels](images/update.settings.channel_menu.png)
+
+And then click **Help -> Check for Update**.
+
+!!! warning WARNING
+    In **nightly** and **alpha** channels sometimes plugins don't have builds for the new platform. (For example if compilation failed.)
+
+    **Keep calm - Consulo do not allow update, if some plugins missed, you will get warning about it.**
+<!--- Editor's note: @Vistall Can you rephrase that please? --->
```
Comment (halotroop2288) [inline docs/platform/settings/platform.and.plugins/update.settings.md]: I need a preliminary review. There is a line in here written by @VISTALL that I don't quite understand.

File: docs/platform/settings/external.services.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Services
+
+- Error Reporting - **Default: Enabled** (Anonymous)<br/>
+	Whether to send anonymous error reporting data to [Consulo.io].<br/>
+	This helps the developer(s) maintain the IDE and fix these errors.<br/>
+	This data **is** anonymous.
+- Developer List (Error Reporting) - **Default: Disabled**<br/>
+	Whether to attach a list of developers to your error reporting data when it is sent to [Consulo.io].<br/>
+	<!--- Why? ---><br/>
```
Comment (halotroop2288) [inline docs/platform/settings/external.services.md]: Why is the developer list asked for?
This would be nice to tell users so they can properly judge whether or not they want to send the data.

Comment (VISTALL) [reply docs/platform/settings/external.services.md]: It's API check for calling develop list from hub. It's only allow if user is already developer. I just show this settings for all - but maybe it will be good, hide it, and show only for developer auth.

It's used to assign developer inside Error Reporter dialog. For example if developer catch exception, and want assign it to another (it can be viewed inside hub, if attached)

File: docs/platform/settings/platform.and.plugins/update.settings.md (hunk 1 above)
Comment (halotroop2288) [reply docs/platform/settings/platform.and.plugins/update.settings.md]: @VISTALL could you please explain this one? Or rephrase it?

**Comment by VISTALL** [comment]
Hello. Thanks for contribution, but for now mkdocs can't serve this docs due YAML error -> section url not supported (https://www.mkdocs.org/user-guide/writing-your-docs/#configure-pages-and-navigation)

```
      - Platform And Plugins: platform/preferences/platform.and.plugins.md  !!!Here problem. Need child node for it
        - Experimental Features: platform/preferences/platform.and.plugins/experimental.features.md
        - Update Settings: platform/preferences/platform.and.plugins/update.settings.md
```

Also due strict mode - moving docs is bad idea (because all links checked).  And moving docs without redirects also bad idea, because it broke old links .

We can use https://github.com/datarobot/mkdocs-redirects plugin for it.

Also new nightly builds(3102+), now have different logic for settings url generate, and need add redirect anyway.

Thanks:)

**Comment by VISTALL** [comment]
Also, root path changed from prefereces -> settings (and need make redirect)

**Comment by halotroop2288** [comment]
It's changed back already...?

**Comment by VISTALL** [comment]
back? I decided after our dialog, change it. As you suggested.

**Comment by halotroop2288** [comment]
I changed most references to "Preferences" in documentation to "Settings" leaving alone parts that specifically mention *both* Preferences *and* Settings for clarity.
